transcom/mymove
# B-21410

## [B-21410](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1014946)
## [INT Review](https://github.com/transcom/mymove/pull/14483)

File count difference comes from:
milmove-cert-bundle.p7b
go.mod
go.sum
migrations/app/secure/20250110214012_homesafeconnect_cert.up.sql (it was added, and then renamed, so the pr counted it)
yarn.lock
package.json

Please double check these, but I figured this would save you some time digging through 100+ files.

## Summary

Adding the IOSHUT and IDSHUT (International Origin and Destination Shuttle Service Items) to the application. They should have the ability to be created, approved, rejected, resubmitted and priced as part of a payment packet.

### How to test

1. Access the Milmove
2. Login as a Customer
3. Create a new move with an HHG
4. Login as a Service Counselor
5. Approve the move details
6. Login as a TOO
7. Approve the shipment and Submit
8. Login as the Prime
9. Update the HHG with dates and weights
10. Add an IOSHUT and IDSHUT service item
11. Login as TOO
12. Reject the service items
13. Using swagger submit the following to mto-service-items/{id}, changing the reServiceCode to match appropriately:
{
  "reServiceCode": "IDSHUT",
  "updateReason": "A reason that differs from the previous reason",
  "modelType": "UpdateMTOServiceItemInternationalShuttle",
  "requestApprovalsRequestedStatus": true
}

15. Login as TOO
16. Verify that the service items are available for approval
17. Approve the IOSHUT and IDSHUT
18. Login as Prime
19. Create a payment request that include IOSHUT and IDSHUT
20. Login as TOO and verify that the payment service items are priced for the requested Payment Packet
